File: Exercise_3/main.py
# --- CLASS: Point2D ---
class Point2D:
    def __init__(self, x=0.0, y=0.0):
        self.__x = x  
        self.__y = y

    # Getters
    def get_x(self): return self.__x
    def get_y(self): return self.__y

    # Setters
    def set_x(self, x): self.__x = x
    def set_y(self, y): self.__y = y

    def set_xy(self, x, y):
        self.__x = x
        self.__y = y

    def get_xy(self):
        return [self.__x, self.__y] 

    def __str__(self):
        return f"({self.__x},{self.__y})"

# --- CLASS: Point3D ---
class Point3D(Point2D):
    def __init__(self, x=0.0, y=0.0, z=0.0):
        super().__init__(x, y)
        self.__z = z

    def get_z(self): return self.__z
    def set_z(self, z): self.__z = z

    def set_xyz(self, x, y, z):
        self.set_xy(x, y) 
        self.set_z(z)     

    def get_xyz(self):
        return [self.get_x(), self.get_y(), self.__z]

    def __str__(self):
        return f"({self.get_x()},{self.get_y()},{self.__z})"

File: Exercise_3/main.java
package Exercise_3;

class Point2D {
    private float x;
    private float y;

    public Point2D() {
        this.x = 0.0f;
        this.y = 0.0f;
    }

    public Point2D(float x, float y) {
        this.x = x;
        this.y = y;
    }

    // Getters & Setters
    public float getX() { return x; }
    public float getY() { return y; }

    public void setX(float x) { this.x = x; } 
    public void setY(float y) { this.y = y; }

    public void setXY(float x, float y) {
        this.x = x;
        this.y = y;
    }

    public float[] getXY() {
        return new float[]{x, y};
    }
    @Override
    public String toString() {
        return "(" + x + "," + y + ")";
    }
}

// --- CLASS: Point3D ---
class Point3D extends Point2D {
    private float z;

    public Point3D() {
        super(0.0f, 0.0f); // Gọi constructor cha
        this.z = 0.0f;
    }

    public Point3D(float x, float y, float z) {
        super(x, y); 
        this.z = z;
    }

    // Getters & Setters
    public float getZ() { return z; }
    public void setZ(float z) { this.z = z; }

    public void setXYZ(float x, float y, float z) {
        setXY(x, y); 
        setZ(z);    
    }

    public float[] getXYZ() {
        return new float[]{getX(), getY(), z};
    }

    @Override
    public String toString() {
        return "(" + getX() + "," + getY() + "," + z + ")";
    }
}

public class main {
    
}
